dot/state/block.go

```diff
@@ -773,10 +773,14 @@ func (bs *BlockState) IsDescendantOf(ancestor, descendant common.Hash) (bool, er
 			return false, fmt.Errorf("getting header: %w", err2)
 		}
 
-		for current := descendantHeader; descendantHeader.Number < ancestorHeader.Number; {
+		for current := descendantHeader; current.Number < ancestorHeader.Number; {
```

Comment:
```suggestion
		for current := descendantHeader; ancestorHeader.Number < current.Number; {
```
The descendant must not have a block number lower than the ancestor